uitest/Pages/DashboardPage.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Copyright (c) Microsoft Corporation and Contributors
+// Licensed under the MIT license.
+
+using DevHome.UITest.Dialogs;
+using DevHome.UITest.Extensions;
+using OpenQA.Selenium.Appium;
+using OpenQA.Selenium.Appium.Windows;
+
+namespace DevHome.UITest.Pages;
+
+/// <summary>
+/// Page model class for the dashboard page
+/// </summary>
+public class DashboardPage : ApplicationPage
+{
+    private WindowsElement AddWidgetButton => Driver.FindElementByAccessibilityId("AddWidgetButton");
+
+    private IReadOnlyCollection<WindowsElement> WidgetItems => Driver.FindElementsByAccessibilityId("WidgetItem");
+
+    public IReadOnlyCollection<WidgetControl> DisplayedWidgets => WidgetItems.Select(i => new WidgetControl(Driver, i)).ToList();
```

Comment:
Nit: These may take a while since it is re-calculated each time. I'd make them methods instead of properties to signal that.